### DISCUSSION TITLE
MVVM pattern and Oxyplot

### DISCUSSION THREAD

**Discussion by NeoStarkX** [oxyplot/oxyplot#1970] [Q&A] [answered]
Hello everybody I hope everything is well with you.
First I want to thank you for providing such a useful tool for us.
I have a question and I'll appreciate your help. I am developing a WPF application with MVVM pattern which means "**View**", "**ViewModel**" and "**Model**" are separated from each other. 
Now I've added **oxyplot** to my "**View**" through NuGet package manager in visual studio 2022 and I can happily add control to my UI. But I have the "**ViewModel**" project that is responsible for Binding stuff and when I look at documentation and examples for Databinding I see that in "**ViewModel**" we have a reference to the "**oxyplot**" namespace. so somehow I have to access the classes defined in "**oxyplot**". It is obvious that I can not add it to my "**ViewModel**" project through the NuGet package manager because it also adds UI stuff to my project which violates the "**MVVM**" pattern. What I have managed to do so far is to download the source codes from GitHub and add them to my "**ViewModel**" project so I can use it there. also, I renamed the "**oxyplot**" namespace in the "**ViewModel**" project to inhibit namespace conflicts that may arise because both "**Oxyplot**" and "**ViewModel**" are referenced in the "**View**" project. the next thing is I came up with a situation in that I had to define a "**Value Converter**" to convert stuff from "**Oxyplot**" in "**ViewModel**" to "**Oxyplot**" in "**View**" for example for the "**PlotModel**" Binding I defined a "**Value Converter**" to convert "**PlotModel**" in my "**ViewModel**" to "**PlotModel**" in "**View**". Now it seems everything is fine and works perfectly.
Now I want to know If my solution is good or bad and if there is any better solution for my scenario.
Thanks in advance.

EDIT: I think adding Oxyplot.Core from NuGet package manager is the Best solution. Right?

**Comment by VisualMelon** [comment, marked as answer]
Sorry for the slow response; I had originally thought I'd misunderstood your question, but my answer would have been "target the OxyPlot.Core package in your ViewModel/Models", and seeing your edit it I would say yes, that is the way to go. I usually keep PlotModels in the ViewModel, as they include interaction information.

**Reply by NeoStarkX** [reply]
Thank you very much for your answer ❤️🙏
